eva-server/src/main/java/uk/ac/ebi/eva/server/ws/VariantWSServerV2.java

```diff
@@ -0,0 +1,257 @@
+            checkParameters(variantCoreString, null, null, species, assembly);
+        } catch (IllegalArgumentException e) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return new Resource<>(setErrorQueryResponse(e.getMessage()));
+        }
+
+        MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName(species + "_" + assembly));
+
+        Optional<VariantWithSamplesAndAnnotation> variantEntity;
+
+        try {
+            variantEntity = getVariantByCoordinatesAndAnnotationVersion(variantCoreString, null, null);
+        } catch (AnnotationMetadataNotFoundException ex) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return new Resource<>(setQueryResponse(ex.getMessage()));
+        } catch (IllegalArgumentException ex) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return new Resource<>(setQueryResponse(ex.getMessage()));
+        }
+
+        List<Variant> variantList = new ArrayList<>();
+        if (!variantEntity.isPresent()) {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            return new Resource<>(setQueryResponse(variantList));
+        }
+
+        Variant variant = new Variant(variantEntity.get().getChromosome(), variantEntity.get().getStart(),
+                variantEntity.get().getEnd(), variantEntity.get().getReference(), variantEntity.get().
+                getAlternate());
+        variant.setIds(variantEntity.get().getIds());
+        variantList.add(variant);
+
+        QueryResult<Variant> queryResult = buildQueryResult(variantList, 1);
+
+        Link annotationLink = new Link(linkTo(methodOn(VariantWSServerV2.class).getAnnotations(variantCoreString,
+                species, assembly, null, null, response)).toUri().toString(), "annotation");
+
+        Link sourcesLink = new Link(linkTo(methodOn(VariantWSServerV2.class).getSources(variantCoreString, species,
+                assembly, null, null, response)).toUri().toString(), "sources");
+
+        List<Link> links = new ArrayList<>();
+        links.add(sourcesLink);
+        links.add(annotationLink);
+        return new Resource<>(setQueryResponse(queryResult), links);
+    }
+
+    private void checkParameters(String variantCoreString, String annotationVepVersion,
+                                 String annotationVepCacheVersion, String species, String assembly) {
+        if (!variantCoreString.contains(":")) {
+            throw new IllegalArgumentException("Please describe a variant as 'sequence:location:reference:alternate'");
+        } else {
+            String[] regionId = variantCoreString.split(":");
+            if(regionId.length!=4) {
+                throw new IllegalArgumentException("VariantCoreString requires 4 fields, " + regionId.length +
+                        "fields were given in the input");
+            }
+            try {
+                Integer.parseInt(regionId[1]);
```

Comment:
We use long numbers to represent a position, and it should be a positive number.
```suggestion
                long position = Long.parseLong(regionId[1]);
                if (position < 0) {
                    throw new IllegalArgumentException("Please provide a positive start position");
                }
```